blackboxopt/optimizers/staged/bohb.py

```diff
@@ -137,11 +146,14 @@ def impute_conditional_data(
             values for inactive variables.
         vartypes: Encoding of the types of the variables: 0 mean continuous, >0 means
             categorical with as many different values, and <0 means ordinal with as many values.
-
+        rng: A random number generator to make the imputation reproducible.
     Returns:
         Numerical representation where all NaNs have been replaced with observed values
         or prior samples.
     """
+    if rng is None:
+        rng = np.random.default_rng()
```

Comment:
```suggestion
    rng = np.random.default_rng(rng)
```
Same as above, I think.

Reply:
PS: Just FYI: I think I can't just "commit suggestion" on your "suggested changes", because those commits won't be signed. They are helpful nevertheless.

Reply:
True, I forgot about that requirement. But if you don't mind, I will keep doing them in the cases where the suggestion is much simpler than trying to describe the change in the text.

Reply:
For sure, please keep doing it, wherever it seems helpful. As a requester, this is super useful as well! Just wanted to explain the reason for not committing suggestion right away... ;-)